### PR TITLE
CADC-12492 - bring neossat2caom2 up-to-date with caom2pipe API, fix p…

### PR DESCRIPTION
**Pull request by SharonGoliath**
…review generation failure for a white/black image.

### CODE REVIEW

File: neossat2caom2/data_source.py [hunk 1]
```diff
@@ -67,202 +67,58 @@
...
-                if dt >= self._start_dt:
-                    temp = f'{root_url}/{d}'
-                    self._logger.debug(f'Adding Day: {temp}')
-                    result[dt].append(temp)
-        return result
+    # There's no timestamp check for the top-level directories when harvesting incrementally because the
+    # lower pages have products retroactively added that don't change the timestamps on the YEAR page.
+    #
+    # VA 23-03-23
+    # We are finally starting to push some of our “advanced” image products to the CSA Open Data.  These
```
Comment (andamian) [inline neossat2caom2/data_source.py]: This justification comments are a kind of requirements traceability but I'm not sure they are useful in the code especially for other developers. Better IMO to comment on the actual functionality of the code.

File: neossat2caom2/tests/test_composable.py [hunk 2]
```diff
@@ -104,90 +102,117 @@
...
+ 
+        test_config.write_to_file(test_config)
+        mc.State.write_bookmark(test_config.state_fqn, test_config.data_sources[0], TEST_START_TIME)
+        with open(test_config.proxy_fqn, 'w') as f:
+            f.write('test content')
+
+        run_mock.return_value = 0
+        try:
+            test_result = composable._run_state()
+        except Exception as e:
```
Comment (andamian) [inline neossat2caom2/tests/test_composable.py]: I'm not sure the `try` block is needed here. Without it, the unexpected exception will propagate out of the test method causing it to fail and report the exception.

File: neossat2caom2/tests/test_composable.py [hunk 3]
```diff
@@ -104,90 +102,117 @@
...
+            logging.error(traceback.format_exc())
+            assert False, f'unexpected exception {e}'
+        assert test_result == 0, 'expected successful test result'
+        assert run_mock.called, 'should be called'
+        args, kwargs = run_mock.call_args
+        test_storage = args[0]
+        assert isinstance(test_storage, NEOSSatName), type(test_storage)
+        assert test_storage.file_name.startswith(
+            'NEOS_SCI'
+        ) and test_storage.file_name.endswith('.fits'), test_storage.file_name
```
Comment (andamian) [inline neossat2caom2/tests/test_composable.py]: If you don't mind regex (`re` needs imported`)
```suggestion
        assert re.match('^NEOS_SCI.*\.fits$', test_storage.file_name) is not None
```

File: neossat2caom2/tests/test_composable.py [hunk 4]
```diff
@@ -104,90 +102,117 @@
...
-    test_end_bookmark = test_state_end.get_bookmark(test_config.bookmark)
-    assert test_end_bookmark == TEST_START_TIME, f'wrong end time max {TEST_START_TIME} end {test_end_bookmark}'
+        os.chdir(tmpdir)
+        test_config.write_to_file(test_config)
+        mc.State.write_bookmark(test_config.state_fqn, test_config.data_sources[0], TEST_START_TIME)
+        with open(test_config.proxy_fqn, 'w') as f:
+            f.write('test content')
+
+        run_mock.return_value = 0
+        try:
```
Comment (andamian) [inline neossat2caom2/tests/test_composable.py]: Same comment about `try` block not necessary

File: neossat2caom2/tests/test_preview_augmentation.py [hunk 5]
```diff
@@ -130,3 +130,34 @@ def test_preview_augmentation(test_config, tmpdir):
...
+    test_builder = nbc.GuessingBuilder(NEOSSatName)
+    test_file_names = ['NEOS_SCI_2023105181520.fits']
+
+    for f_name in test_file_names:
+        test_obs = mc.read_obs_from_file(
+            f'{test_main_app.TEST_DATA_DIR}/{f_name.replace(".fits", "").replace("NEOS_SCI_", "")}.expected.xml'
+        )
+        test_fqn = f'/test_files/{f_name}'
+        if not os.path.exists(test_fqn):
+            assert False, f'missing {test_fqn}'
```
Comment (andamian) [inline neossat2caom2/tests/test_preview_augmentation.py]: `if` can be part of the `assert`

**Review by andamian — APPROVED** [review]

File: neossat2caom2/data_source.py (hunk 1 above)
Comment (SharonGoliath) [reply neossat2caom2/data_source.py]: Started a page in Confluence to track these types of comments.

File: neossat2caom2/tests/test_composable.py (hunk 2 above)
Comment (SharonGoliath) [reply neossat2caom2/tests/test_composable.py]: Removed from test_composable.py.

File: neossat2caom2/tests/test_composable.py (hunk 3 above)
Comment (SharonGoliath) [reply neossat2caom2/tests/test_composable.py]: Done.

File: neossat2caom2/tests/test_composable.py (hunk 4 above)
Comment (SharonGoliath) [reply neossat2caom2/tests/test_composable.py]: Removed.

File: neossat2caom2/tests/test_preview_augmentation.py (hunk 5 above)
Comment (SharonGoliath) [reply neossat2caom2/tests/test_preview_augmentation.py]: Done.

**Pull request merged**